### ISSUE TITLE
Immediate replies quote the context

### ISSUE THREAD

**Issue by progval** [matrix-org/matrix-appservice-irc#1525]
**Describe the bug**
Replying to someone normally does not quote the message being replied to if it's the last one someone wrote, but it looks like this doesn't always work

**To Reproduce**
Not sure. Earlier in #libera-matrix at Libera, I saw this:

```
10:10:38 <+halfy> anyway anyway, I'll add it to the thought pile
[two IRC users joining and quitting repeatedly]
12:20:02 <pwr22> <halfy> "anyway anyway, I'll add it to..." <- Is yours also as large as Everest? 😆
[two people joining and quitting]
13:11:46 <+halfy> <pwr22> "Is yours also as large as..." <- hey, i like a challenge :p
```

**Expected behavior**

Should simply be this:

```
10:10:38 <+halfy> anyway anyway, I'll add it to the thought pile
[two IRC users joining and quitting repeatedly]
12:20:02 <pwr22> halfy: Is yours also as large as Everest? 😆
[two people joining and quitting]
13:11:46 <+halfy> pwr22: hey, i like a challenge :p
```

**Comment by Half-Shot** [comment]
@tadzik you might know best for this one?

**Comment by tadzik** [comment]
> Replying to someone normally does not quote the message being replied to if it's the last one someone wrote

This is actually time-based: see [config.sample.yaml](https://github.com/matrix-org/matrix-appservice-irc/blob/develop/config.sample.yaml#L577):

```
    # format of replies sent shortly after the original message
    shortReplyTemplate: "$NICK: $REPLY"
    # format of replies sent a while after the original message
    longReplyTemplate: "<$NICK> \"$ORIGINAL\" <- $REPLY"
    # how much time needs to pass between the reply and the original message to switch to the long format
    shortReplyTresholdSeconds: 300
```

We can consider relaxing it a little and checking whether this is indeed the last time someone spoke – the intention behind it being time-based was the fact that someone might have last spoken a long time ago and the original context is long lost behind a wall of joins, parts and unrelated conversations. Plus it was easier to implement :)

**Comment by emorrp1** [comment]
Yes, last timeline message behaviour would be smarter than just a timeout, not everyone sits in extremely busy channels where there's going to be cross-talk every 5mins. Bonus points for checking last timeline message *for $NICK* since that shortReplyTemplate implies context there anyway.
